pkg/cluster/cluster.go

```diff
@@ -35,23 +35,62 @@ type Peer struct {
 	stopc chan struct{}
 }
 
+var (
+	pushPullInterval = 5 * time.Second
+	gossipInterval   = 5 * time.Second
+)
+
 // PeerType describes a peer's role in the cluster.
 type PeerType string
 
 // Constants holding valid PeerType values.
 const (
-	PeerTypeStore = "store"
-	PeerTypeQuery = "query"
+	PeerTypeStoreGCS     = "store-gcs"
+	PeerTypeStoreSidecar = "store-sidecar"
+	PeerTypeQuery        = "query"
 )
 
+// PeerState contains state for the peer.
+type PeerState struct {
+	Type    PeerType
+	APIAddr string
+
+	Metadata PeerMetadata
+}
```

Comment:
Why? what's the benefit of it?

Reply:
Mh, I figured it might be easier for forward/backward compatibility later. But I might be wrong – also might still be too early either way.
Size would be another argument – but certainly not with our current amount of data.

Summed up, probably not important (yet) after all.

Reply:
Ok,  let's do that someday, but not in this PR

Reply:
👍